src/main/java/seedu/address/logic/parser/todolist/AddTaskCommandParser.java

```diff
@@ -0,0 +1,70 @@
+package seedu.address.logic.parser.todolist;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PRIORITY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.AddCommand;
+import seedu.address.logic.commands.todolist.AddTaskCommand;
+import seedu.address.logic.parser.ArgumentMultimap;
+import seedu.address.logic.parser.ArgumentTokenizer;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.ParserUtil;
+import seedu.address.logic.parser.Prefix;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.task.Date;
+import seedu.address.model.task.Priority;
+import seedu.address.model.task.Task;
+import seedu.address.model.task.TaskName;
+
+/**
+ * Parses input arguments and creates a new AddTaskCommand object
+ */
+public class AddTaskCommandParser implements Parser<AddTaskCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddCommand
+     * and returns an AddCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddTaskCommand parse(String args) throws ParseException {
+        Task task;
+        ArgumentTokenizer tokenizer = new ArgumentTokenizer(args, PREFIX_NAME, PREFIX_TAG, PREFIX_PRIORITY,
+                PREFIX_DATE);
+        ArgumentMultimap argMultimap = tokenizer.tokenize();
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME)
+            || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
+        }
+        TaskName taskName = ParserUtil.parseTaskName(argMultimap.getValue(PREFIX_NAME).get());
+        task = new Task(taskName);
+
+        if (argMultimap.getValue(PREFIX_TAG).isPresent()) {
+            Tag taskTag = ParserUtil.parseTag(argMultimap.getValue(PREFIX_TAG).get());
```

Comment:
Perhaps we could implement the tag feature of Tasks as a Set<Tag> so that users will be able to add more than 1 tag. Based on this current implementation, users will only be allowed to add one tag. However, it is fine for now and we can make the necessary changes later during the milestone.

Reply:
Agreed, I will update this in the future